bm-persona/Libraries/LibraryDetailViewController.swift

```diff
@@ -0,0 +1,362 @@
+        nameLabel.adjustsFontSizeToFitWidth = true
+        
+        let addressIcon = UIImageView()
+        addressIcon.image = #imageLiteral(resourceName: "Walk")
+        card.addSubview(addressIcon)
+        addressIcon.translatesAutoresizingMaskIntoConstraints = false
+        addressIcon.leftAnchor.constraint(equalTo: card.layoutMarginsGuide.leftAnchor).isActive = true
+        addressIcon.topAnchor.constraint(equalTo: nameLabel.layoutMarginsGuide.bottomAnchor, constant: 20).isActive = true
+        addressIcon.widthAnchor.constraint(equalTo: card.layoutMarginsGuide.widthAnchor, multiplier: 0.06).isActive = true
+        addressIcon.contentMode = .scaleAspectFit
+        
+        
+        let addressLabel = UILabel()
+        if let longAddress = library!.campusLocation {
+            if let ind = longAddress.range(of: "Berkeley")?.upperBound {
+                let newAdress = longAddress[..<ind]
+                addressLabel.text = String(newAdress)
+            } else {
+                addressLabel.text = longAddress
+            }
+        }
+        card.addSubview(addressLabel)
+        addressLabel.font = Font.thin(12)
+        addressLabel.numberOfLines = 2
+        addressLabel.textColor = UIColor.black
+        addressLabel.translatesAutoresizingMaskIntoConstraints = false
+        addressLabel.topAnchor.constraint(equalTo: nameLabel.layoutMarginsGuide.bottomAnchor, constant: 20).isActive = true
+        addressLabel.leftAnchor.constraint(equalTo: addressIcon.layoutMarginsGuide.rightAnchor, constant: 8).isActive = true
+        //addressLabel.widthAnchor.constraint(equalTo: card.layoutMarginsGuide.widthAnchor, multiplier: 0.5).isActive = true
+        addressLabel.rightAnchor.constraint(equalTo: card.layoutMarginsGuide.rightAnchor, constant: -160).isActive = true
+        addressLabel.adjustsFontSizeToFitWidth = true
+        
+        
+        let phoneLabel = UILabel()
+        phoneLabel.text = library!.phoneNumber
+        card.addSubview(phoneLabel)
+        phoneLabel.font = Font.thin(12)
+        phoneLabel.textColor = UIColor.black
+        phoneLabel.translatesAutoresizingMaskIntoConstraints = false
+        phoneLabel.topAnchor.constraint(equalTo: addressLabel.layoutMarginsGuide.bottomAnchor, constant: 20).isActive = true
+        phoneLabel.leftAnchor.constraint(equalTo: card.layoutMarginsGuide.leftAnchor).isActive = true
+        phoneLabel.widthAnchor.constraint(equalTo: card.layoutMarginsGuide.widthAnchor, multiplier: 0.5).isActive = true
+        phoneLabel.adjustsFontSizeToFitWidth = true
+        
+        //walking distance
+        if let libLat = library!.latitude, let libLong = library!.longitude {
+            let dist = calcDistance(lib_Lat: libLat, lib_Long: libLong)
+            distLabel.text = String(format: "%.1f", dist) + " miles away"
+            
+        } else {
+            distLabel.text = "distance not availible"
+        }
```

Comment:
Use HasLocation protocol's getDistanceToUser function instead. Also, are we displaying "distance not available" or are we not displaying anything? Should make sure this is consistent with the table.